packages/eslint-plugin/src/rules/no-deprecated-components/createNoDeprecatedComponentsRule.ts

```diff
@@ -112,10 +129,34 @@ export function createNoDeprecatedComponentsRule(
                             messageId: "migration",
                             node,
                         });
+                    } else if (isOpeningElement(node.parent)) {
+                        // check <DeprecatedComponent withDeprecatedProp={...}> syntax
+                        const deprecatedProp = node.parent.attributes.find(
+                            attribute =>
+                                attribute.type === TSESTree.AST_NODE_TYPES.JSXAttribute &&
+                                attribute.name.type === TSESTree.AST_NODE_TYPES.JSXIdentifier &&
+                                deprecatedToNewComponentMapping[`${node.name}.${attribute.name.name}`] != null,
+                        );
+                        if (deprecatedProp !== undefined) {
```

Comment:
nit: can unnest if we return early instead
```suggestion
                        if (deprecatedProp === undefined) {
                            return;
                        }
```